mantidproject/mantid
# ISIS Powder Prevent division by zero in Vanadium

**Description of work.**
Prevented division by zero and very low values when dividing by vanadium by cropping the workspace where the vanadium is less than 1% of its maximum

**To test:**
Download:
[Test.zip](https://github.com/mantidproject/mantid/files/2277619/Test.zip)

unzip it, update the paths in the config file, then run
```
from isis_powder import Gem

config_file_path = r"Path\to\Gem_config_example.yaml"
Gem_example = Gem(config_file=config_file_path,user_name="test")
Gem_example.create_vanadium(input_mode="Individual",first_cycle_run_no=85105,texture_mode=True)
Gem_example.focus(input_mode="Individual",run_number="85105",texture_mode=True)
```
Plot 85105-ResultTOF_160 85105-ResultD_160 and spline_bank_160. at the point where the spline approximately reaches zero, TOF and D should both have no values.

Fixes #23214 
---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
